backend/src/main/java/com/education/project/persistence/MatiereRepository.java

```diff
@@ -80,4 +80,71 @@ public void closeConnexion(){
             throw new DataBaseException("La création de la matière n'a pas pu être réalisée.");
         }
     }//insert()
+
+    /**
+     * Cette fonction permet de modifier une matière.
+     * @param matiere La matière à modifier.
+     * @return La matière modifiée.
+     * @throws DataBaseException
+     */
+    public Optional<Matiere> modifier(Matiere matiere) throws DataBaseException {
+        String request = "UPDATE matiere SET nom = ?, couleurFond = ?, couleurPolice = ?, volumeHoraire = ?, description = ?, modificationDate = ? WHERE id = ? ";
+        try{
+            PreparedStatement preparedStatement = this.connexion.prepareStatement(request);
+            preparedStatement.setString(1, matiere.getNom());
+            preparedStatement.setString(2, matiere.getCouleurFond());
+            preparedStatement.setString(3, matiere.getCouleurPolice());
+            preparedStatement.setString(4, matiere.getVolumeHoraire());
+            preparedStatement.setString(5, matiere.getDescription());
+            preparedStatement.setDate(6, new java.sql.Date(matiere.getModificationDate().getTime()));
```

Comment:
On va pouvoir régler le problème de la date côté mysql définitivement grâce à cet article : 

https://www.dariawan.com/tutorials/java/java-sql-date-java-sql-time-and-java-sql-timestamp/

new java.sql.Date -> new java.sql.Timestamp

Il faudra faire la même modification pour l'insertion de matière. De plus, au niveau du fichier script.sql il faudra,
remplacer DATETIME par TIMESTAMP pour creationDate et modificationDate. 

Tu peux détruire la table matiere sur mysql (heidisql) et la reconstruire grâce au scriptSQL (ou tout refaire avec l'IHM).